src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -596,6 +596,56 @@ private static void StelemRef(Array array, int index, object? obj)
             StelemRef_Helper(ref element, elementType, obj);
         }
 
+#if TARGET_64BIT
+        [DebuggerHidden]
+        [StackTraceHidden]
+        [DebuggerStepThrough]
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        private static ref object? LdelemaRefNint(Array array, nint index, void* type)
+        {
+            // this will throw appropriate exceptions if array is null or access is out of range.
+            ref object? element = ref Unsafe.As<ArrayElement[]>(array)[index].Value;
+            void* elementType = RuntimeHelpers.GetMethodTable(array)->ElementType;
+
+            if (elementType == type)
+                return ref element;
+
+            return ref ThrowArrayMismatchException();
+        }
+
+        [DebuggerHidden]
+        [StackTraceHidden]
+        [DebuggerStepThrough]
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        private static void StelemRefNint(Array array, nint index, object? obj)
+        {
+            // this will throw appropriate exceptions if array is null or access is out of range.
+            ref object? element = ref Unsafe.As<ArrayElement[]>(array)[index].Value;
+            void* elementType = RuntimeHelpers.GetMethodTable(array)->ElementType;
+
+            if (obj == null)
+                goto assigningNull;
+
+            if (elementType != RuntimeHelpers.GetMethodTable(obj))
+                goto notExactMatch;
+
+            doWrite:
+                WriteBarrier(ref element, obj);
+                return;
+
+            assigningNull:
```

Comment:
There is only one branch to this, does it need to be a labeled jump? In fact, I don't see the reason for any of these labels. Is there a benefit?